File: api-ruby/samples/server/hosted.rb
require 'sinatra/base'
require_relative 'config'
module HostedRoutes
  def self.registered(app)
    app.get '/hosted' do
      send_file File.join(settings.public_folder, 'hosted.html')
    end

    app.get '/hosted-launch/:provider' do
      provider = params[:provider]
      redirectUrl = params[:redirectUrl]
      verificationProfileId = ENV['TRINSIC_VERIFICATION_PROFILE_ID']
      req = TrinsicApi::CreateHostedProviderSessionRequest.new({
        redirect_url: redirectUrl,
        provider: provider,
        verification_profile_id: verificationProfileId
      })

      opts = {
        create_hosted_provider_session_request: req
      }
      result = TrinsicServices::SESSIONS.create_hosted_provider_session(opts)
      redirect result.launch_url
    end
  end
end

File: api-python/samples/server/hosted.py
from fastapi import Request, APIRouter, Depends, HTTPException
from fastapi.responses import JSONResponse
from pydantic import BaseModel
from trinsic_api.api.sessions_api import SessionsApi, CreateHostedProviderSessionRequest
from fastapi.responses import RedirectResponse
import os
import json

hostedRouter = APIRouter()
@hostedRouter.get("/hosted")
async def redirect():
    return RedirectResponse(url="/hosted.html")


@hostedRouter.post("/create-hosted-session/{provider_id}")
async def launch(request: Request, provider_id: str, sessions_api: SessionsApi = Depends()):
    request = CreateHostedProviderSessionRequest(redirect_url = request.query_params.get("redirectUrl"), verification_profile_id=os.getenv("TRINSIC_VERIFICATION_PROFILE_ID"), provider=provider_id)

    result = sessions_api.create_hosted_provider_session(create_hosted_provider_session_request=request)
    
    return JSONResponse(content=json.loads(json.dumps(result.to_dict(), default=json_serial)))
def json_serial(obj):
    """JSON serializer for objects not serializable by default."""
    if isinstance(obj, (datetime, date)):
        return obj.isoformat()
    raise TypeError(f"Type {type(obj)} not serializable")